compiler/src/builder.rs

```diff
@@ -373,28 +377,145 @@ impl CellAllocator {
 		}
 	}
 
-	fn allocate(&mut self, location: Option<TapeCell>, size: usize) -> Result<TapeCell, String> {
+	fn allocate(
+		&mut self,
+		location: Option<TapeCell>,
+		size: usize,
+		method: u8,
```

Comment:
This method type should be an enum, but I think for this sprint it's probably fine, something to refactor next sprint maybe